plugin/src/main/groovy/com/novoda/staticanalysis/internal/findbugs/FindbugsConfigurator.groovy

```diff
@@ -1,50 +1,51 @@
 package com.novoda.staticanalysis.internal.findbugs
 
-import com.novoda.staticanalysis.StaticAnalysisExtension
-import com.novoda.staticanalysis.internal.Violations
+import com.novoda.staticanalysis.EvaluateViolationsTask
+import com.novoda.staticanalysis.internal.CodeQualityConfigurator
+import org.gradle.api.Action
 import org.gradle.api.Project
-import org.gradle.api.Task
 import org.gradle.api.plugins.quality.FindBugs
 import org.gradle.api.plugins.quality.FindBugsExtension
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.JavaExec
 import org.gradle.internal.logging.ConsoleRenderer
 
-class FindbugsConfigurator {
+class FindbugsConfigurator extends CodeQualityConfigurator<FindBugs, FindBugsExtension> {
 
-    void configure(Project project, Violations violations, StaticAnalysisExtension extension, Task evaluateViolations) {
-        extension.ext.findbugs = { Closure config ->
-            project.apply plugin: QuietFindbugsPlugin
-            List<String> excludes = []
-            configureExtension(project.extensions.findByType(FindBugsExtension), excludes, config)
-            project.afterEvaluate {
-                configureAndroidIfNeeded(project)
-                project.tasks.withType(FindBugs) { FindBugs findBugs ->
-                    configureTask(project, findBugs, evaluateViolations, violations, excludes)
-                }
-            }
-        }
+    FindbugsConfigurator(Project project, EvaluateViolationsTask evaluateViolations) {
+        super(project, evaluateViolations.maybeCreate('Findbugs'), evaluateViolations)
     }
 
-    private void configureExtension(FindBugsExtension extension, List<String> excludes, Closure config) {
-        extension.with {
-            toolVersion = '3.0.1'
-            ext.exclude = { String filter -> excludes.add(filter) }
-            config.delegate = it
-            config()
-        }
+    @Override
+    protected String getToolName() {
+        'findbugs'
+    }
+
+    @Override
+    protected Object getToolPlugin() {
+        QuietFindbugsPlugin
+    }
+
+    @Override
+    protected Class<FindBugsExtension> getExtensionClass() {
+        FindBugsExtension
     }
 
-    private void configureAndroidIfNeeded(Project project) {
-        boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
-        boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
-        if (isAndroidApp || isAndroidLib) {
-            def variants = isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants
-            configureAndroid(project, variants)
+    @Override
+    protected Class<FindBugs> getTaskClass() {
+        FindBugs
+    }
+
+    @Override
+    protected Action<FindBugsExtension> getDefaultConfiguration() {
+        new Action<FindBugsExtension>() {
```

Comment:
Yeah, I had `{ -> }` at one point, but I figured someone would have complained during the review. I bet on the wrong horse I guess 🏇